File: Todos/C/BEE 1929.c
/*
  autor: Vinist021;
  data: 26/01/2025;
  nome: Triângulo;
*/

#include <stdio.h>

int triverif(int a, int b, int c)
{
	if(a + b > c && a + c > b && b + c > a)
		return 0;
	else
		return 1;
}

int main() {
	
	int n1, n2, n3, n4;
	
	scanf("%d %d %d %d", &n1, &n2, &n3, &n4);
	
	if(triverif(n1, n2, n3) == 0 || triverif(n1, n2, n4) == 0||
	   triverif(n1, n3, n4) == 0 || triverif(n2, n3, n4) == 0)
		printf("S\n");
	else
		printf("N\n");
	
	return 0;
}

File: Todos/Python/BEE 1929.py
#   autor: Vinist021;
#   data: 26/01/2025;
#   nome: Triângulo;

import itertools

def verificatriangulo(a, b, c):
    return a + b > c and a + c > b and b + c > a

A, B, C, D = map(int, input().split())

tamanhos = [A, B, C, D]

for comb in itertools.combinations(tamanhos, 3):
    if verificatriangulo(*comb) == True:
        print('S')
        break
else:
    print('N')
